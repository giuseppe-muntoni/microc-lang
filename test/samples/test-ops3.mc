void printb(bool val)
{
  if(val)
    print(1);
  else
    print(0);
}

int main()
{
  print_float(9.9900 + 2.15);
  print_float(3.0 - 1.5);
  print_float(1.1 * 2);
  print_float(100.45 / 2.34);
  print_float(99.99999);
  printb(1.1 == 2.2);
  printb(0.99999999 == 1);
  printb(0.999 == 1);
  print_float(99.99999);
  printb(1.1 != 2.2);
  printb(1.85 != 1.85);
  print_float(99.99999);
  printb(1.1 < 2.2);
  printb(2.2 < 1.1);
  print_float(99.99999);
  printb(1.9 <= 2);
  printb(1.1 <= 1);
  return 0;
}