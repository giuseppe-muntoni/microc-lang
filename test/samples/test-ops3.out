12.1400
1.5000
2.2000
42.9273
100.0000
0
1
0
100.0000
1
0
100.0000
1
0
100.0000
1
0
